src/systems.rs

```diff
@@ -112,6 +114,35 @@ impl<'a> System<'a> for CollisionSystem {
                 }
             }
         }
+
+
+        // Collision of Enemies to the Base
+        // This might be worth refactoring since it's nearly identical to code above
```

Comment:
Yeah, I think I would prefer to try to merge these two loops. I'm thinking you can just remove the restriction on projectile/base from iteration. That way all collisions will be sent out. It'll send two events for collisions, but that's okay. We can handle that in the attacker system since it checks for collisions from Attacker -> Defender, not the other way around.